container.go

```diff
@@ -838,6 +838,32 @@ func (c *Client) RemoveContainer(opts RemoveContainerOptions) error {
 	return nil
 }
 
+// PutContainerArchiveOptions is the set of options that can be used when
+// uploading an archive into a container.
+//
+// See https://goo.gl/Ss97HW for more details.
+type PutContainerArchiveOptions struct {
+	InputStream          io.Reader `json:"-" qs:"-"`
+	Container            string    `json:"-" qs:"-"`
+	Path                 string    `qs:"path"`
+	NoOverwriteDirNonDir bool      `qs:"noOverwriteDirNonDir"`
+}
+
+// PutContainerArchive uploads a tar archive to be extracted to a path in the
+// filesystem of the container.
+//
+func (c *Client) PutContainerArchive(opts PutContainerArchiveOptions) error {
+	if opts.Container == "" {
```

Comment:
See the comment on https://github.com/fsouza/go-dockerclient/pull/356/files, not sure if we want to do this...if Container is required, I think our standard is to make it a separate parameter, not an option (especially since you are ignoring it for json).